crates/emitter/src/ast_emitter.rs

```diff
@@ -99,16 +101,32 @@ impl<'alloc, 'opt> AstEmitter<'alloc, 'opt> {
                 self.scope_stack.leave_lexical(&mut self.emit);
             }
             Statement::BreakStatement { .. } => {
-                return Err(EmitError::NotImplemented("TODO: BreakStatement"));
+                BreakEmitter {
+                    jump: JumpKind::Goto,
+                }
+                .emit(self);
             }
             Statement::ContinueStatement { .. } => {
                 return Err(EmitError::NotImplemented("TODO: ContinueStatement"));
             }
             Statement::DebuggerStatement { .. } => {
                 return Err(EmitError::NotImplemented("TODO: DebuggerStatement"));
             }
-            Statement::DoWhileStatement { .. } => {
-                return Err(EmitError::NotImplemented("TODO: DoWhileStatement"));
+            Statement::DoWhileStatement { block, test, .. } => {
+                self.loop_stack.open_loop(&mut self.emit);
+
+                self.emit_statement(block)?;
```

Comment:
Please use `ExpressionEmitter`, to emit `pop` or `set_rval`.
here and other places that emits loop body.

Reply:
this was wrong